certbot-nginx/certbot_nginx/_internal/configurator.py

```diff
@@ -719,7 +719,11 @@ def enhance(self, domain, enhancement, options=None):
 
     def _has_certbot_redirect(self, vhost, domain):
         test_redirect_block = _test_block_from_block(_redirect_block_for_domain(domain))
-        return vhost.contains_list(test_redirect_block)
+        redirect_block_for_server = [['\n    ', 'return', '301', 'https://$host$request_uri']]
```

Comment:
This seems specific to the config in #6466. We might want something more generic that can detect the same directive with any tabbing.

And since this is a regular list, I think you need the whitespace information between tokens. Might be able to reuse some existing code in `_redirect_block_for_domain(domain)` here!

Reply:
The tabbing is actually obselete now since I use `nginxparser.UnspacedList`.

I also added a unit Test if `_has_certbot_redirect` returns True if a custom block exists.